### PR TITLE
feat: prepare for @zendeskgarden npm publish

### PR DESCRIPTION
**Pull request by jzempel**
* [x] **BREAKING CHANGE?** 

## Description

Package location has moved to the npm registry under the `@zendeskgarden` scope. See the README for installation/upgrade details.

## Detail

Additionally:

* apply license header to all JS files
* fill-out README npm versions & dependency checks
* rename `yarn build:styleguide` to `yarn build:demo` (more straightforward)

## Checklist

* [x] :ok_hand: design updates are Garden Designer approved (add the
      designer as a reviewer)
* [x] :nail_care: view component styling is based on a Garden CSS
  component
* [x] :globe_with_meridians: Styleguidist demo is up-to-date (`yarn start`)
* [x] :arrow_left: renders as expected with reversed (RTL) direction
* [x] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -9,22 +9,15 @@ cache:
...
-  - npm config set //zdrepo.jfrog.io/zdrepo/api/npm/npm/:username $ARTIFACTORY_USERNAME
-  - npm config set //zdrepo.jfrog.io/zdrepo/api/npm/npm/:email $ARTIFACTORY_EMAIL
-  - npm config set //zdrepo.jfrog.io/zdrepo/api/npm/npm/:always-auth true
-
 install: yarn --frozen-lockfile
 
 script:
   - yarn lint
-  - yarn test:all
+  - yarn test:all --runInBand
```
Comment (austin94) [inline .travis.yml]: Is there a reason we moved the `--runInBand` up to Travis? `yarn test` is what the majority of development would be using. I think having that abstracting it out could make it more difficult once we enable `--coverage`.

File: utils/travis/publish-gh-pages.js [hunk 2]
```diff
@@ -1,24 +1,15 @@
...
-      console.log(error);
-      throw error;
-    }
 
-    console.log('GH-Pages upload complete.');
-  }
-);
+const ghPages = require('gh-pages');
+
+ghPages.publish('demo', {
```
Comment (austin94) [inline utils/travis/publish-gh-pages.js]: I know we originally decided to remove the `add: true` option for ghPages, but I think we forgot about the "pre-publish" steps we do sometime to shared changes.

Without the `add` option those pre-published files will be wiped out. I think leaving those files around will be helpful.

File: .travis.yml (hunk 1 above)
Comment (jzempel) [reply .travis.yml]: I guess it depends on what you're doing whether you're using `yarn test` or `yarn test:all`. But when running `test:all` its preferable to take advantage of a multi-core computer (18s with `--runInBand`; 8s without). I'm sure we can finesse `--coverage` when we get to that.

File: utils/travis/publish-gh-pages.js (hunk 2 above)
Comment (jzempel) [reply utils/travis/publish-gh-pages.js]: Ok, let's leave that as a conversation following this PR.

**Pull request merged**